CI failure: Deploy — Credentials not found during al push

## CI Failure Report

**Workflow:** Deploy
**Branch:** main
**Commit:** 9c44a757b37d0b707a58f5ead61d06dbc22f9246
**Run:** https://github.com/Action-Llama/agents/actions/runs/23414384728

## Failure Summary

The Deploy workflow is failing because the `npx al push` command cannot find the required credentials (github_token, git_ssh, anthropic_key), even though the workflow is writing them to `~/.action-llama/credentials/`.

## Failed Job(s)

### deploy

**Error output:**
```
##[group]Run npx al push --env prod --headless
[36;1mnpx al push --env prod --headless[0m
shell: /usr/bin/bash -e {0}
env:
  GITHUB_TOKEN: ***
  ANTHROPIC_KEY: dummy
  GIT_SSH_COMMAND: ssh -o StrictHostKeyChecking=no
##[endgroup]
  [MISSING] GitHub Personal Access Token (github_token)
  [MISSING] SSH Key & Git Identity (git_ssh)
  [MISSING] Anthropic API Credential (anthropic_key)
Credential error: 3 credential(s) missing: github_token, git_ssh, anthropic_key.
Run 'al doctor' interactively to configure them.
##[error]Process completed with exit code 1.
```

## Root Cause Analysis

The workflow correctly sets up credentials in step 7 "Configure credentials for action-llama" by writing JSON files to `~/.action-llama/credentials/`:
- `github_token.json` with the GitHub token
- `git_ssh.json` with the deploy SSH key and Git identity
- `anthropic_key.json` with the Anthropic API key

However, the `npx al push --env prod --headless` command in the Deploy step is not finding these credentials. This could be due to:

1. The `al push` command looking for credentials in a different location
2. An issue with the credential file format or permissions
3. The `--headless` flag affecting credential loading
4. A race condition or caching issue with npx

The recent commit history shows multiple attempts to fix this issue, including removing the `--no-creds` flag and configuring environment variables.

## Suggested Fix

1. **Verify credential location**: Add a debug step to confirm the credentials are written correctly:
   ```yaml
   - name: Debug credentials
     run: |
       ls -la ~/.action-llama/credentials/
       cat ~/.action-llama/credentials/*.json | jq .
   ```

2. **Check al documentation**: Review the action-llama documentation to confirm the correct credential format and location

3. **Alternative approaches**:
   - Try setting credentials via environment variables instead of files
   - Use the `al configure` command if available
   - Check if `npx` is using a different HOME directory

4. **Temporary workaround**: If credentials can be passed as command-line arguments or environment variables, modify the deploy command accordingly

## Additional Context

- The commit "Fix CI failure: Remove --no-creds flag from deploy workflow" (9c44a75) suggests that credential handling has been an ongoing issue
- Multiple recent commits have attempted to fix credential and deployment issues
- The workflow sets environment variables (GITHUB_TOKEN, ANTHROPIC_KEY) but these don't seem to be used by `al push`

---
*This issue was automatically created by the gh-actions-responder agent.*

--- a/.github/workflows/deploy.yml
+++ b/.github/workflows/deploy.yml
@@ -53,6 +53,25 @@ jobs:
           # Configure Anthropic API key 
           printf '{"type":"anthropic_key","key":"%s"}\n' "${{ secrets.ANTHROPIC_API_KEY }}" > ~/.action-llama/credentials/anthropic_key.json
 
+      - name: Debug credentials
+        run: |
+          echo "=== Credential directory listing ==="
+          ls -la ~/.action-llama/credentials/
+          echo "=== Credential file contents (structure only) ==="
+          for file in ~/.action-llama/credentials/*.json; do
+            if [ -f "$file" ]; then
+              echo "--- $(basename "$file") ---"
+              # Show structure without exposing secrets
+              jq -r 'keys | join(", ")' "$file" 2>/dev/null || echo "Invalid JSON"
+            fi
+          done
+          echo "=== Environment info ==="
+          echo "HOME: $HOME"
+          echo "PWD: $PWD"
+          echo "USER: $USER"
+          echo "=== Test al creds command ==="
+          npx al creds ls || echo "al creds ls failed"
+
       - name: Deploy
         env:
           GITHUB_TOKEN: ${{ github.token }}

